cmd/main.go

```diff
@@ -93,6 +78,11 @@ func main() {
 	var probeAddr string
 	var secureMetrics bool
 	var enableHTTP2 bool
+	var batchDebounceInterval time.Duration
```

Comment:
naming nit: the definition for debounce is commonly referring to the time _after_ the first event, which I don't think applies here because we're using this as a ticker interval.

I'd suggest `batchInterval` is clearer - or perhaps `maxBatchInterval` if you want to capture that batches may be sent more frequently.